pkg/engine/reconciler.go

```diff
@@ -71,10 +149,24 @@ func (e *Engine) deleteResource(resource core.Resource, explicit bool) bool {
 	}
 
 	for _, res := range upstreamNodes {
-		e.deleteResource(res, false)
+		explicit := false
+		for _, reflectResource := range reflectResources {
+			if res == reflectResource {
+				explicit = true
+				continue
+			}
+		}
+		e.deleteResource(res, explicit, false)
 	}
 	for _, res := range downstreamNodes {
-		e.deleteResource(res, false)
+		explicit := false
+		for _, reflectResource := range reflectResources {
+			if res == reflectResource {
+				explicit = true
+				continue
+			}
+		}
+		e.deleteResource(res, explicit, false)
 	}
 	return true
 }
```

Comment:
Definitely could use some unit tests on these to help wrap my head around what these reconciler functions are supposed to do. Some more godocs as well, where appropriate.